### PR TITLE
Add vesting info

### LINKED ISSUE #320: Add page on vesting
Include what vesting is, how it works, and "lazy vesting claims" information

### PR DESCRIPTION
**Pull request by lsaether**
closes #320

### CODE REVIEW

File: docs/learn-DOT.md [hunk 1]
```diff
@@ -58,7 +115,7 @@ _Warning: Mainnet DOT tokens are not transferrable until mainnet launch, expecte
 Testnet dots are freely available now - see below for various ways to obtain them.
 
 ## Testnet DOTs
-
+ubstrate-telemetry-
```
Comment (lsaether) [inline docs/learn-DOT.md]:
```suggestion
```

File: docs/mirror-maintain-guides-how-to-validate-kusama.md [hunk 2]
```diff
@@ -82,12 +82,34 @@ command INSTEAD of the previous one:
...
+```
+
+
 ### Building and Installing the `polkadot` Binary
 
 You will need to build the `polkadot` binary from the [paritytech/polkadot](https://github.com/paritytech/polkadot)
 repository on GitHub using the source code available in the **v0.7** branch.
 
-You should generally use the latest **0.7.x** tag.  At the time of writing, this was **0.7.16**, but you should review the output from the "git tag" command (`git tag | grep "$v\0\.7"`) to see a list of all the potential 0.7 releases.  You should replace `v0.7.16` with the latest build (i.e., the highest number).
+You should generally use the latest **0.7.x** tag.  At the time of writing, this was **0.7.20**, but you should review the output from the "git tag" command (`git tag | grep "$v\0\.7"`) to see a list of all the potential 0.7 releases.  You should replace `v0.7.20` with the latest build (i.e., the highest number). You can also find the latest Kusama version on the [release](https://github.com/paritytech/polkadot/releases) tab.
```
Comment (joepetrowski) [inline docs/mirror-maintain-guides-how-to-validate-kusama.md]:
```suggestion
You should generally use the latest **0.7.x** tag.  At the time of writing, this was **0.7.28**, but you should review the output from the "git tag" command (`git tag | grep "$v\0\.7"`) to see a list of all the potential 0.7 releases.  You should replace `v0.7.28` with the latest build (i.e., the highest number). You can also find the latest Kusama version on the [release](https://github.com/paritytech/polkadot/releases) tab.
```

File: docs/mirror-maintain-guides-how-to-validate-kusama.md [hunk 3]
```diff
@@ -96,7 +118,7 @@ You should generally use the latest **0.7.x** tag.  At the time of writing, this
 git clone https://github.com/paritytech/polkadot.git
 cd polkadot
 git tag | grep "$v\0\.7"
-git checkout v0.7.16
+git checkout v0.7.20
```
Comment (joepetrowski) [inline docs/mirror-maintain-guides-how-to-validate-kusama.md]:
```suggestion
git checkout v0.7.28
```

File: docs/mirror-maintain-guides-how-to-validate-kusama.md [hunk 4]
```diff
@@ -133,64 +147,11 @@ if you do not want to start in validator mode right away.
...
-```sh
-./target/release/polkadot --pruning=archive
-```
-
 Depending on the size of the chain when you do this, this step may take anywhere from a few minutes to a few hours.
 
 If you are interested in determining how much longer you have to go, your server logs (printed to STDOUT from the
 `polkadot` process) will tell you the latest block your node has processed and verified. You can then compare that to
-the current highest block via [Telemetry](https://telemetry.polkadot.io/#list/Kusama%20CC2) or the
+the current highest block via [Telemetry](https://telemetry.polkadot.io/#list/Kusama%20CC3) or the
```
Comment (joepetrowski) [inline docs/mirror-maintain-guides-how-to-validate-kusama.md]:
```suggestion
the current highest block via [Telemetry](https://telemetry.polkadot.io/#list/Kusama) or the
```

File: docs/mirror-maintain-guides-how-to-validate-kusama.md [hunk 5]
```diff
@@ -291,7 +256,7 @@ Submit this extrinsic and you are now ready to start validating.
 
 ## Validate
 
-To verify that your node is live and synchronized, head to [Telemetry](https://telemetry.polkadot.io/#list/Kusama%20CC1)
+To verify that your node is live and synchronized, head to [Telemetry](https://telemetry.polkadot.io/#/Kusama%20CC3)
```
Comment (joepetrowski) [inline docs/mirror-maintain-guides-how-to-validate-kusama.md]:
```suggestion
To verify that your node is live and synchronized, head to [Telemetry](https://telemetry.polkadot.io/#/Kusama)
```

File: docs/mirror-maintain-guides-how-to-validate-kusama.md [hunk 6]
```diff
@@ -232,6 +193,10 @@ account.
 
 ## Set Session Keys
 
+> **Note:** The session keys are consensus critical, so if you are not sure if your node has the current session keys
+> that you made the `setKeys` transaction for then simply run through the process of generating and setting new ones
+> using the `rotateKeys` method below. Better safe than sorry!
```
Comment (joepetrowski) [inline docs/mirror-maintain-guides-how-to-validate-kusama.md]: Actually we have these two RPC methods that check if you have keys: `hasKey` and `hasSessionKeys`

https://polkadot.js.org/api/substrate/rpc.html#haskey-publickey-bytes-keytype-text-bool

File: docs/mirror-maintain-guides-how-to-validate-kusama.md [hunk 7]
```diff
@@ -360,5 +313,5 @@ over the network.
 If you have Docker installed, you can use it to start your validator node without needing to build the binary. You can do this with a simple one line command:
 
 ```sh
-$ docker run parity/polkadot:v0.5.0 --validator --name "name on telemetry"
+$ docker run parity/polkadot:v0.7.20 --validator --name "name on telemetry"
```
Comment (joepetrowski) [inline docs/mirror-maintain-guides-how-to-validate-kusama.md]:
```suggestion
$ docker run parity/polkadot:v0.7.28 --validator --name "name on telemetry"
```

File: docs/learn-DOT.md [hunk 8]
```diff
@@ -49,6 +53,59 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
 
 DOTs will have the ability to be bonded for a duration of time in order to add a new parachain to the network. The DOTs will be locked during their bonding period and will be released back to the account that bonded them after the duration of the bond has elapsed and the parachain is removed.
 
+### Vesting
+
+DOTs that are allocated to the Polkadot team or awarded to teams as part of 
+Web3 Foundation's grants program will have a vesting schedule attached. Vested
```
Comment (joepetrowski) [inline docs/learn-DOT.md]: I don't think this is necessary for the concept of vesting.

File: docs/learn-DOT.md [hunk 9]
```diff
@@ -49,6 +53,59 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
...
+balances are classified as a separate abstraction to the other locks that are
+placed on your balance. This means that although vested tokens cannot be transferred,
+they can still be used for voting in governance or staked in the staking system.
```
Comment (joepetrowski) [inline docs/learn-DOT.md]: Undeclared "this".

Nonetheless, the fact that vesting is a separate lock isn't what lets you use vesting funds in governance/staking, that's just a property of locks.

File: docs/learn-DOT.md [hunk 10]
```diff
@@ -49,6 +53,59 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
...
+```rust
+pub struct VestingInfo<Balance, BlockNumber> {
+	/// Locked amount at genesis.
+	pub locked: Balance,
+	/// Amount that gets unlocked every block after `starting_block`.
+	pub per_block: Balance,
+	/// Starting block for unlocking(vesting).
+	pub starting_block: BlockNumber,
+}
+```
```
Comment (joepetrowski) [inline docs/learn-DOT.md]: I'm opposed to showing code like this. It's out of context and does nothing to make the idea more concrete. You can easily explain this with natural language. Vesting is a simple concept and any user, no matter how technical, should be able to understand it from this section.

File: docs/learn-DOT.md [hunk 11]
```diff
@@ -49,6 +53,59 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
...
+- The second way is through a transaction type available in the "Vesting" pallet,
+`vested_transfer`. The vested transfer function allows anyone to create a 
+vesting schedule with a transfer of funds, with the caveat that the account
+for which the vesting schedule will be created does not already have one. To
+call vested transfer successfully you must transfer at least `MinVestedTransfer`
+funds, which is specified as a chain constant.
+
+#### Lazy Vesting
+
+Like [lazy payouts](learn-lazy-payouts), vesting is _lazy_. This means that
```
Comment (joepetrowski) [inline docs/learn-DOT.md]:
```suggestion
Like [lazy payouts](learn-lazy-payouts), vesting is _lazy_, which means that
```

File: docs/learn-DOT.md [hunk 12]
```diff
@@ -49,6 +53,59 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
...
+someone must explicitly call either `vest` to vest themselves (the caller) or
```
Comment (joepetrowski) [inline docs/learn-DOT.md]: Should say that calling `vest` just updates the lock to reflect the new (lower) amount that is vesting.

File: docs/learn-DOT.md [hunk 13]
```diff
@@ -49,6 +53,59 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
...
+for which the vesting schedule will be created does not already have one. To
+call vested transfer successfully you must transfer at least `MinVestedTransfer`
+funds, which is specified as a chain constant.
+
+#### Lazy Vesting
+
+Like [lazy payouts](learn-lazy-payouts), vesting is _lazy_. This means that
+someone must explicitly call either `vest` to vest themselves (the caller) or 
+`vest_other` to unlock some vested funds from another target account. These
+extrinsics are available as part of the "Vesting" pallet.
```
Comment (joepetrowski) [inline docs/learn-DOT.md]:
```suggestion
extrinsics are available as part of the Vesting pallet.
```

File: docs/learn-DOT.md [hunk 14]
```diff
@@ -49,6 +53,59 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
...
+
+Vested funds that are "vested" will remain locked until one of the two above
```
Comment (joepetrowski) [inline docs/learn-DOT.md]:
```suggestion
Vested funds will remain locked until one of the two above
```

**Review by joepetrowski — CHANGES_REQUESTED** [review]

**Comment by lsaether** [comment]
@joepetrowski addressed your comments. Also made the updates to the validator guide.

File: docs/learn-DOT.md [hunk 15]
```diff
@@ -49,6 +53,47 @@ DOTs will be used to facilitate the consensus mechanism that underpins Polkadot.
 
 DOTs will have the ability to be bonded for a duration of time in order to add a new parachain to the network. The DOTs will be locked during their bonding period and will be released back to the account that bonded them after the duration of the bond has elapsed and the parachain is removed.
 
+### Vesting
+
+DOTs may have a lock placed on them to account for vesting funds. Like other types
+of locks, these funds cannot be transferred but can be used in other parts of
+the protocol such as voting in governance or being staked as a validator or nominator.
+
+Vesting funds are on a linear release schedule and unlock a constant amount of
```
Comment (joepetrowski) [inline docs/learn-DOT.md]:
```suggestion
Vesting funds are on a linear release schedule and unlock a constant number of
```

**Review by joepetrowski — APPROVED** [review]

**Pull request merged**